api/datastore/sql/migratex/migrate.go

```diff
@@ -0,0 +1,337 @@
+package migratex
```

Comment:
Looks straightforward, so the only thing that bothers me is migration collecting, for migrations we have I'd like to make this framework go and collect every migration it has and then apply them step by step, similar to what I did in my PR. This is what i felt when saw the test case for the migrations. At this moment, it looks quite complex.

So, we want people to create the migrations and then just append them to the global list of migrations and then let the framework sort them, and apply, we shouldn't expose much logic to users except the transaction object sqlx.Tx.

I see that we have necessary logic for sorting those, so there's nothing much left to do.

Reply:
> Looks straightforward, so the only thing that bothers me is migration collecting, for migrations we have I'd like to make this framework go and collect every migration it has and then apply them step by step, similar to what I did in my PR. This is what i felt when saw the test case for the migrations. At this moment, it looks quite complex.

could add a step-wise instead of all up or all down; just trying to keep the surface area small since we were only ever going all the way up or down (mattes has 5-6 ways to run migrations, it's all noise). this is what you mean, correct?

> So, we want people to create the migrations and then just append them to the global list of migrations and then let the framework sort them, and apply, we shouldn't expose much logic to users except the transaction object sqlx.Tx.

I suppose it's pretty flexible since we just take a `[]Migration`, the global list is one way to get them in there. does this comment imply some change you'd like to see? it's not clear.

thank you for reviewing.